source/Kepler-System/CompositeSystem.class.st

```diff
@@ -1,60 +1,133 @@
+"
+I'm a composite system, a system containing another systems (called subsystems). I'm the one responsible of coordinating the startUp and shutDown of a whole subsystems combination.
+"
 Class {
 	#name : #CompositeSystem,
 	#superclass : #SystemImplementation,
 	#instVars : [
-		'systems'
+		'subsystems',
+		'installationOptional'
 	],
-	#category : 'Kepler-System'
+	#category : #'Kepler-System'
 }
 
 { #category : #'instance creation' }
-CompositeSystem class >> of: aSystemsCollection [
+CompositeSystem class >> installedBy: anInstallation [
 
-	^self new initializeOf: aSystemsCollection
+	^self new initializeInstalledBy: anInstallation
+]
+
+{ #category : #'subsystem lookup' }
+CompositeSystem >> >> anInterfaceKey [
+
+	^ self systemImplementing: anInterfaceKey
+]
+
+{ #category : #'private - accessing' }
+CompositeSystem >> asInterface: anInterfaceKey [
```

Comment:
Checking the use in `systemImplementing:ifFound:ifNone:` I think a message starting with as... is confusing because we are not transforming self, but rather the collaborator.
A better name might be `interfaceFor:` or `interfaceRelatedTo:`